iceoryx_utils/include/iceoryx_utils/internal/cxx/vector.inl

```diff
@@ -291,44 +288,32 @@ inline const T& vector<T, Capacity>::operator[](const uint64_t index) const noex
 template <typename T, uint64_t Capacity>
 inline T& vector<T, Capacity>::front() noexcept
 {
-    if (empty())
-    {
-        std::cerr << "Attempting to access the front of an empty vector!" << std::endl;
-        std::terminate();
-    }
+    // No empty vector shall be accessed
+    cxx::Expects(!empty() && "Attempting to access the front of an empty vector");
     return at(0);
 }
 
 template <typename T, uint64_t Capacity>
 inline const T& vector<T, Capacity>::front() const noexcept
 {
-    if (empty())
-    {
-        std::cerr << "Attempting to access the front of an empty vector!" << std::endl;
-        std::terminate();
-    }
+    // No empty vector shall be accessed
```

Comment:
I think we should use const cast here and return the const casted value of the non const `front()` method to avoid code duplication. Otherwise the doubled error handling can become easily out of sync.